parsl/tests/test_providers/test_submiterror_deprecation.py

```diff
@@ -0,0 +1,19 @@
+import pytest
+
+from parsl.providers.errors import SubmitException
+
+
+@pytest.mark.local
+def test_submit_execption_task_name_deprecation():
+    """This tests the deprecation warning of task_name in SubmitException
+    """
+    j = "the_name"
+
+    ex = SubmitException(j, "m")
+
+    # the new behaviour
+    assert ex.job_name == j
```

Comment:
Were you so inclined, I like to use `randomstring()` (a fixture) for tests like this.  Sort of a "keeping honest" promise.  Here's an [example implementation](https://github.com/funcx-faas/funcX/blob/main/compute_endpoint/tests/conftest.py#L149-L156) from Globus Compute.  Then, this might look like:

```python
j = f"some_job_{randomstring()}"
```

Reply:
I'll bring this into Parsl in a separate PR - it looks good to have around